draft-ietf-tls-esni.md

```diff
@@ -432,36 +432,35 @@ identified by `ECHConfig.cipher_suites`. Once a suitable configuration is found,
 the client selects the cipher suite it will use for encryption. It MUST NOT
 choose a cipher suite not advertised by the configuration.
 
-Next, the client constructs the ClientHelloOuter message just as it does a
-standard ClientHello, with the exception of the following rules:
-
-1. It MUST offer to negotiate TLS 1.3 or above.
-1. It MUST include an "encrypted_client_hello" extension with a payload
-   constructed as described below.
-1. The value of `ECHConfig.public_name` MUST be placed in the "server_name"
-   extension.
-1. It MUST NOT include the "pre_shared_key" extension. (See
-   {{flow-clienthello-malleability}}.)
-
-The client then constructs the ClientHelloInner message just as it does a
+Next, the client constructs the ClientHelloInner message just as it does a
 standard ClientHello, with the exception of the following rules:
 
 1. It MUST NOT offer to negotiate TLS 1.2 or below. Note this is necessary to
    ensure the backend server does not negotiate a TLS version that is
    incompatible with ECH.
 1. It MUST NOT offer to resume any session for TLS 1.2 and below.
-1. It MAY offer any other extension in the ClientHelloOuter except those that
-   have been incorporated into the ClientHelloInner as described in
-   {{outer-extensions}}.
-1. It MAY copy any other field from the ClientHelloOuter except
-   ClientHelloOuter.random. Instead, It MUST generate a fresh
-   ClientHelloInner.random using a secure random number generator. (See
-   {{flow-client-reaction}}.)
 1. It SHOULD contain TLS padding {{!RFC7685}} as described in {{padding}}.
+
+The client then constructs the ClientHelloOuter message just as it does a
+standard ClientHello, with the exception of the following rules:
+
+1. It MUST offer to negotiate TLS 1.3 or above.
+1. Any extensions compressed as described in {{outer-extensions}} must match
```

Comment:
This rule could be made more clear. Suggestion:
> Any extension that appears in the payload of the "outer_extensions" extension of the ClientHelloInner MUST appear in the ClientHelloOuter.

Reply:
I didn't want to talk about outer_extensions specifically appearing in ClientHelloInner because of #331. The ClientHelloInner doesn't really have outer_extensions, only the payload of the ECH extension. We haven't settled yet on text there, or a processing order of actually encrypting things (I'll need to fiddle with that as part of #323), so for now I figure citing the actual process was the least ambiguous.

The distinction is important because of which is actually used in the handshake. (Transcript and whatnot.)

Reply:
Then I would just leave an "OPEN ISSUE" so we know to fix this later.

Reply:
Done.